odoo/addons/test_read_group/tests/test_web_group_expand.py

```diff
@@ -0,0 +1,359 @@
+from odoo.tests import common
+
+
+class TestGroupExpand(common.TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(**cls.env.context, read_group_expand=True))
+        cls.Model = cls.env['test_read_group.on_selection']
+
+    def test_none(self):
+        self.Model.create({'value': 1})
+        self.Model.create({'value': 2})
+        self.Model.create({'value': 3})
+
+        groups = self.Model.web_read_group([], ['state'], ['__count', 'value:sum'])['groups']
+        self.assertEqual(groups, [
+            {
+                'state': 'a',
+                '__count': 0,
+                'value:sum': False,
+                '__extra_domain': [('state', '=', 'a')],
+            },
+            {
+                'state': 'b',
+                '__count': 0,
+                'value:sum': False,
+                '__extra_domain': [('state', '=', 'b')],
+            },
+            {
+                'state': False,
+                '__count': 3,
+                'value:sum': 6,
+                '__extra_domain': [('state', '=', False)],
+            },
+        ])
+
+    def test_partial(self):
+        self.Model.create({'state': 'a', 'value': 1})
+        self.Model.create({'state': 'a', 'value': 2})
+        self.Model.create({'value': 3})
+
+        groups = self.Model.web_read_group([], ['state'], ['__count', 'value:sum'])['groups']
+        self.assertEqual(groups, [
+            {
+                'state': 'a',
+                '__count': 2,
+                'value:sum': 3,
+                '__extra_domain': [('state', '=', 'a')],
+            },
+            {
+                'state': 'b',
+                '__count': 0,
+                'value:sum': False,
+                '__extra_domain': [('state', '=', 'b')],
+            },
+            {
+                'state': False,
+                '__count': 1,
+                'value:sum': 3,
+                '__extra_domain': [('state', '=', False)],
+            },
+        ])
+
+    def test_full(self):
+        self.Model.create({'state': 'a', 'value': 1})
+        self.Model.create({'state': 'b', 'value': 2})
+        self.Model.create({'value': 3})
```

Comment:
To test all states filled (partial tested above), shouldn't that be:
```suggestion
        self.Model.create({'state': 'c', 'value': 3})
```

Reply:
I want to test False value actually.